clicache/integration-test2/SerializationTests.cs

```diff
@@ -510,5 +823,119 @@ public void PutGetCustomSerializableTypes()
                 cache.Close();
             }
         }
-    }
-}
+
+		[Fact]
+		public void CompositeClassWithClassAsKey()
```

Comment:
Good addition.   Does a non-composite class used as a key exercise the same code paths?  if not, be great to add a test for that too...

Reply:
I agree we should have a separate ClassAsKey. It would only use ReadInternalObject if the class being used as the key had a member that was a container (i.e., list, vector, dictionary). I had it that way originally.

Reply:
Added a new test ClassAsKey.